edb/server/net_worker.py

```diff
@@ -363,6 +364,10 @@ async def _gc(tenant: edbtenant.Tenant, expires_in: statypes.Duration) -> None:
             for db in tenant.iter_dbs():
                 if db.name == defines.EDGEDB_SYSTEM_DB:
                     continue
+                if not tenant.is_database_connectable(db.name):
+                    # Don't run the net_worker if the database is not
+                    # connectable, e.g. being dropped
+                    continue
```

Comment:
Should we be checking in `_delete_requests` also?

(More generally, in the medium term I think we need to figure out design patterns/interfaces for worker tasks that get these things right by default)

Reply:
Fixed for now - let's see if it fixes the test

Reply:
Nope, still failing.